apsi_diploma/diploma_app/models.py

```diff
@@ -1,3 +1,88 @@
 from django.db import models
+from viewflow.models import Process
+from django.conf import settings
+import datetime
 
-# Create your models here.
+
+BACHELOR_OF_SCIENCE = "BS"
+MASTERS = "M"
+DOCTORAL = "D"
+SCIENTIFIC = "S"
+PAPER_TYPE_CHOICES = [
+    (BACHELOR_OF_SCIENCE, "Bachelor of Science dissertation"),
+    (MASTERS, "Master's dissertation"),
+    (DOCTORAL, "Doctoral dissertation"),
+    (SCIENTIFIC, "Scientific paper"),
+]
+
+
+class ScientificPublishingProcess(Process):
+
+    title = models.CharField(max_length=500)
+    description = models.CharField(max_length=5000)
+    file = models.FileField()
+    paper_type = models.CharField(
+        max_length=2,
+        choices=PAPER_TYPE_CHOICES,
+    )
+    organizational_unit = models.CharField(max_length=256)
+
+
+class DissertationProcess(Process):
+
+    UNSATISFACTORY = 2.0
+    SATISFACTORY = 3.0
+    SATISFACTORY_PLUS = 3.5
+    GOOD = 4.0
+    GOOD_PLUS = 4.5
+    VERY_GOOD = 5.0
+    GRADE_CHOICES = (
+        (UNSATISFACTORY, "Unsatisfactory"),
+        (SATISFACTORY, "Satisfactory"),
+        (SATISFACTORY_PLUS, "Satisfactory plus"),
+        (GOOD, "Good"),
+        (GOOD_PLUS, "Good plus"),
+        (VERY_GOOD, "Very good"),
+    )
+
+    supervisor = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        on_delete=models.CASCADE,
+        related_name="supervisor",
+        blank=True,
+        null=True,
+    )
+    topic_title = models.CharField(max_length=500)
+    topic_description = models.CharField(max_length=5000)
+    topic_approved = models.BooleanField(default=False)
+    dissertation_file = models.FileField()
+    dissertation_accepted = models.BooleanField(default=False)
+    paper_type = models.CharField(
+        max_length=2,
+        choices=PAPER_TYPE_CHOICES,
+    )
+    supervisor_review = models.FileField()
```

Comment:
Should it be a file field or rather a text field? Let's align on that, I am not too sure.

Reply:
I think text field will be better.